sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -90,8 +91,8 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
         table.schema.findNestedField(Seq(colName), resolver = conf.resolver)
           .map(_._2.dataType)
           .getOrElse {
-            throw QueryCompilationErrors.alterColumnCannotFindColumnInV1TableError(
-              quoteIfNeeded(colName), table)
+            throw QueryCompilationErrors.unresolvedColumnError(
```

Comment:
> Can we reuse `UNRESOLVED_COLUMN.WITH_SUGGESTION`?

For the scenario of `ALTER COLUMN`, since there is one table, I think it is OK; for `resolveFieldNames` in `Analyzer`, I think it would be better if the table can be explicitly specified.